s4/clarity/artifact.py

```diff
@@ -213,5 +214,89 @@ def reagent_label_name(self, reagent_label_name):
         reagent_label = self.make_subelement_with_parents("./reagent-label")
         reagent_label.set("name", reagent_label_name)
 
+    @lazy_property
+    def demux(self):
+        # type: () -> ArtifactDemux
+        """
+        Provides access to the 'demux' endpoint added in Clarity 5.1.
+
+        Using this property with an earlier version of Clarity will result in
+        a 404 Not Found error.
+
+        :type: ArtifactDemux
+        """
+        return ArtifactDemux(self.lims, self.uri + "/demux")
+
     def _get_attach_to_key(self):
         return self.type, ""
+
+
+class DemuxArtifact(WrappedXml):
+    """
+    Corresponds to the 'demux-artifact' type in the 'artifact' namespace of the Clarity API.
+    """
+    artifact = subnode_link(Artifact, ".")
+    reagent_labels = subnode_element_list(ReagentLabel, "reagent-labels", "reagent-label", readonly=True)
+
+    @property
+    def demux(self):
+        # type: () -> DemuxDetails
+        """
+        The element `DemuxDetails` from subnode 'demux'
+
+        :type: DemuxDetails
+        """
+        demux_root = self.xml_root.find("./demux")
+        if (demux_root is None):
+            if (len(self.samples) == 1 and len(self.reagent_labels) > 1):
```

Comment:
Should probably be a comment why this is in here. In the moment we both know but I suspect it will be confusing to future us or anyone else.

Reply:
Good point. I'll add one.

Reply:
While I was writing this, I realized that our workaround doesn't cover the case where the sub-pool contains one labeled artifact and one unlabeled artifact from the same sample. Might wait for someone to report that that's actually a problem, though. :)